source/_posts/Planetscape-a-dystopian-escape-game-for-36C3.md

```diff
@@ -0,0 +1,189 @@
+- (For the future) include other game masters to expand the main story or create sub-stories.
+
+### Introducing THC
+In order to orchestrate all different components of the game, Social Dist0rtion Protocol (yes, still us) developed a game engine called [Treasure Hunt Creator][0] (**THC** for short) that, given a story structured in directories, is able to:
+
+- Build and deploy the front-end application.
+- Build, deploy and initialize the game smart contract on the Ethereum Network.
+- Encrypt and deploy the chapters on IPFS.
+
+[THC][0] played a crucial role in the creation of the game, allowing us to meet our high standards in terms of development experience. But THC was more than that. Fast prototyping and automatic deploying gave us the piece of mind we needed during our last hours of work.
+
+### Zero Knowledge Proof
+As mentioned earlier, every time a correct password is found, a transaction is sent to the smart contract managing the game. Ethereum transactions are public, and we don't want players to cheat sniffing passwords submitted by others. To avoid that, we implemented a cheap zero-knowledge proof scheme. Roughly speaking, the scheme is the following:
+
+1. Given a valid `password`, a player's `player_address`:
+2. The dApp generates `seed = keccak256(password)`.
+3. The dApp generates `chapter_private_key` from `seed`.
+4. The dApp generates `signature = sign(address, chapter_private_key)`
+5. The dApp calls the `submit` method of the smart sontract passing `signature`.
+6. The smart contract checks the current `chapter` for `player_address` in its storage.
+7. The smart contract loads the `chapter_address` for `chapter`.
+8. The smart contracts calculates `signing_address` by doing an `ecrecover(player_address, signature)`.
+9. If `signing_address == chapter_address`, then the smart contract updates the current chapter for the player, moving them to the next chapter.
+
+More information about the [creation][10] and [verification][11] of the proof can be found in the source code.
+
+## Marketing
+Given the importance of **operation Planetscape** we wanted to onboard as many people as possible. We are not acquainted with custom and traditions of the early 2000<sup id="a1">[1](#f1)</sup> (remember, we are from the future) so we tried three orthogonal strategies to engage with attendees of the Congress.
+
+### Personal Envelope
+Before reaching the Congress we handcrafted 35 personal letters and embedded them in envelopes<sup id="a2">[2](#f2)</sup>. Each letter contained the text with the call to action to open the website, start the game, and enter the password for *Chapter 0*. Three SDP members carried them around the congress to personally onboard attendees of 36C3. Handing the envelope was **not** a trivial task and required interaction to assess the willingness of the prospect to participate to the game. Remember, we had a limited number of envelopes, so we had to play them carefully.
+
```

Comment:
do you have a picture of the letter to embed here?

Reply:
Not of the envelop :( but I have a couple of more pictures form the congress to share.